File: Data Structures & Algorithms/combination-target-sum/submission-3.go
import "slices"
func combinationSum(nums []int, target int) [][]int {
    var rec func(int, int, []int)
    ans := make([][]int, 0)

    rec = func(i int, s int, arr []int){
        if s == target{
            t := make([]int, len(arr))
            copy(t, arr)
            ans = append(ans, t)
            return
        }
        if s > target{
            return
        }
        if i >= len(nums){
            return
        }

        arr = append(arr, nums[i])
        rec(i, s+nums[i], arr)

        arr = slices.Delete(arr, len(arr)-1, len(arr))
        rec(i+1, s, arr)
    }
    rec(0,0,[]int{})

    return ans
}

File: Data Structures & Algorithms/generate-parentheses/submission-3.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:

        ans = []
        def rec(o,c,arr):
            print(o,c,arr)
            if o == n and c == n:
                ans.append(arr)
                return
            if c > o:
                return
            
            if o < n:
                rec(o+1,c,arr+"(")
                
            if c < n:
                rec(o,c+1,arr+")")
                
        rec(0,0,"")

        return ans

        
        